x/gateway/keeper/msg_server_stake_gateway.go

```diff
@@ -0,0 +1,91 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdkerrors "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/pokt-network/poktroll/x/gateway/types"
+)
+
+func (k msgServer) StakeGateway(
+	goCtx context.Context,
+	msg *types.MsgStakeGateway,
+) (*types.MsgStakeGatewayResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	logger := k.Logger().With("method", "StakeGateway")
+	logger.Info(fmt.Sprintf("About to stake gateway with msg: %v", msg))
+
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
+
+	// Check if the gateway already exists or not
+	var err error
+	var coinsToDelegate sdk.Coin
+	gateway, isGatewayFound := k.GetGateway(ctx, msg.Address)
+	if !isGatewayFound {
+		logger.Info(fmt.Sprintf("Gateway not found. Creating new gateway for address %s", msg.Address))
+		gateway = k.createGateway(ctx, msg)
+		coinsToDelegate = *msg.Stake
+	} else {
+		logger.Info(fmt.Sprintf("Gateway found. Updating gateway stake for address %s", msg.Address))
+		currGatewayStake := *gateway.Stake
+		if err = k.updateGateway(ctx, &gateway, msg); err != nil {
+			return nil, err
+		}
+		coinsToDelegate = (*msg.Stake).Sub(currGatewayStake)
+	}
+
+	// Retrieve the address of the gateway
+	gatewayAddress, err := sdk.AccAddressFromBech32(msg.Address)
+	if err != nil {
+		logger.Error(fmt.Sprintf("could not parse address %s", msg.Address))
```

Comment:
I feel like we should **either** log **or** return errrors, but **not both**.

Reply:
After thinking about this for a bit, I think my suggestion would be to change these to `logger.Debug()` calls instead of `#Error()`. The node operator shouldn't have to fill their logs with every invalid request; however, I believe the utility of (and perhaps original intention behind) having this log is to aid in debugging.
```suggestion
		// TODO_TECHDEBT(#384): determine whether to continue using cosmos logger for debug level.
		logger.Debug(fmt.Sprintf("could not parse address %s", msg.Address))
```